README.md

```diff
@@ -256,6 +256,28 @@ const client = new Client(authClient);
 const result = await client.createInvoice({amount: 1000});
 ```
 
+#### Handling refresh token 
+Access tokens do expire. If an access token is about to expire, this library will automatically use a refresh token to retrieve a fresh one. Utilising the tokens event is a simple approach to guarantee that you always save the most recent tokens:
+
+```js
+const token = loadTokenForUser(); // {access_token: string, refresh_token: string, expires_at: number}
+const authClient = new auth.OAuth2User({
+  client_id: process.env.CLIENT_ID,
+  callback: "http://localhost:8080/callback",
+  scopes: ["invoices:read", "account:read", "balance:read", "invoices:create", "invoices:read", "payments:send"],
+  token: token
+});
+
+// listen to the tokens event
+authClient.on('tokens', (tokens) => {
+  if (tokens.refresh_token) {
```

Comment:
why do we need a check here? the event should always provide a valid token.

Why log twice?

We should add another event for failed token refresh, right?

Reply:
can we do it in the same tokens event, that refreshAccessToken do. ```Promise<{ token: Token }>``` return this, if token refresh is done. else return null. the user stores the tokens if not. it handles the error by relaunching oauth flow? 


if token refresh is failed we will be still returning expired token that already has in ```this.token```. handling error and getting toking, two different methods related to same purpose,  wouldn't  be little inconsistent?